internal/backends/backend.go

```diff
@@ -79,6 +86,94 @@ func (bc *backendContract) Close() {
 	resource.Untrack(bc, bc.token)
 }
 
+// CreateUser stores a new user in the given database and Backend.
+func CreateUser(ctx context.Context, b Backend, mechanisms *types.Array, dbName, username, password string) error {
+	credentials, err := MakeCredentials(mechanisms, username, password)
+	if err != nil {
+		return err
+	}
+
+	id := uuid.New()
+	saved := must.NotFail(types.NewDocument(
+		"_id", dbName+"."+username,
+		"credentials", credentials,
+		"user", username,
+		"db", dbName,
+		"roles", types.MakeArray(0),
+		"userId", types.Binary{Subtype: types.BinaryUUID, B: must.NotFail(id.MarshalBinary())},
+	))
+
+	adminDB, err := b.Database("admin")
+	if err != nil {
+		return err
+	}
+
+	users, err := adminDB.Collection("system.users")
+	if err != nil {
+		return err
+	}
+
+	_, err = users.InsertAll(ctx, &InsertAllParams{
+		Docs: []*types.Document{saved},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// MakeCredentials creates a document with credentials for the chosen mechanisms.
+func MakeCredentials(mechanisms *types.Array, username, userPassword string) (*types.Document, error) {
+	credentials := types.MakeDocument(0)
+
+	if mechanisms.Len() == 0 {
+		return nil, lazyerrors.New("mechanisms field must not be empty")
+	}
+
+	// TODO: if the optional field mechanisms is nil, create a user with all SCRAM mechanisms,
+	// this is how it works in MongoDB.
+
+	iter := mechanisms.Iterator()
+	defer iter.Close()
+
+	for {
+		var v any
+		_, v, err := iter.Next()
+
+		if errors.Is(err, iterator.ErrIteratorDone) {
+			break
+		}
+
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		switch v {
+		case "PLAIN":
+			credentials.Set("PLAIN", must.NotFail(password.PlainHash(userPassword)))
+		case "SCRAM-SHA-1":
+			hash, err := password.SCRAMSHA1Hash(username, userPassword)
+			if err != nil {
+				return nil, err
+			}
+
+			credentials.Set("SCRAM-SHA-1", hash)
+		case "SCRAM-SHA-256":
+			hash, err := password.SCRAMSHA256Hash(userPassword)
+			if err != nil {
+				return nil, NewError(ErrorCodeStringProhibited, err)
```

Comment:
There is no guarantee any error here is due to bad password... 

I think prohibited password should return appropriate error handled in createUser, but for --setup best I think best it can do is just fail setup with appropriate error. I don't think this is the best place to replace error.